src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -19,29 +19,37 @@
 
 package org.elasticsearch.index.query;
 
+import com.google.common.base.Preconditions;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.BooleanClause.Occur;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.lucene.search.Queries;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 
 import java.io.IOException;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Locale;
 import java.util.Map;
+import java.util.Objects;
 
 /**
  * SimpleQuery is a query parser that acts similar to a query_string
  * query, but won't throw exceptions for any weird string syntax.
  */
-public class SimpleQueryStringBuilder extends QueryBuilder {
+public class SimpleQueryStringBuilder extends QueryBuilder<SimpleQueryStringBuilder> {// TODO Doesn't have a boost value so far
     private Map<String, Float> fields = new HashMap<>();
     private String analyzer;
-    private Operator operator;
-    private final String queryText;
     private String queryName;
+    private Operator operator;
+    private String queryText;
```

Comment:
can we keep mandatory fields final? now that we extend Writeable instead of Streamable we should be able to do that.